swagger_server/controllers/ciupload_controller.py

```diff
@@ -1,48 +1,93 @@
-import connexion
-from datetime import date, datetime
-from typing import List, Dict
-from six import iteritems
-from ..util import deserialize_date, deserialize_datetime
-
-
-def activate_id_put(id):
+##############################################################################
+#                                                                            #
+#   Collection Instruments Upload                                            #
+#   License: MIT                                                             #
+#   Copyright (c) 2017 Crown Copyright (Office for National Statistics)      #
+#                                                                            #
+##############################################################################
+from flask import request, jsonify, make_response
+from .collectioninstrument import CollectionInstrument
+from ons_ras_common.ons_decorators import validate_jwt
+
+collection_instrument = CollectionInstrument()
+
+
+#
+# /status/{id}
+#
+@validate_jwt(['ci:read', 'ci:write'], request)
+def status_id_get(id):
     """
-    Activate batch
-    Mark all items in the batch as available
-    :param id: Collection exercise identifier
+    Get upload status
+    Return a count of items thus far and total in batch
+    :param id: Survey identifier
     :type id: str
 
     :rtype: None
     """
-    return 'do some magic!'
+    code, msg = collection_instrument.status(id)
+    return make_response(jsonify(msg), code)
 
 
-def clear_batch_id_delete(id):
+#
+# /define_batch/{id}/{count}
+#
+@validate_jwt(['ci:read', 'ci:write'], request)
+def define_batch_id_count_post(id, count):
     """
-    Clear a batch
-    Clear down a batch definition, useful for testing
-    :param id: Collection exercise identifier
+    Specify the size of a batch
+    Associate a service and ce with an item count
+    :param id: Survey identifier
     :type id: str
+    :param count: Number of items in batch
+    :type count: int
 
     :rtype: None
     """
-    return 'do some magic!'
+    code, msg = collection_instrument.define_batch(id, count)
+    return make_response(jsonify(msg), code)
 
 
-def define_batch_id_count_post(id, count):
+#
+# /upload/{id}/{file}
+#
+@validate_jwt(['ci:read', 'ci:write'], request)
+def upload_id_file_post(id, file, files=None):
     """
-    Specify the size of a batch
-    Associate a service and ce with an item count
-    :param id: Collection exercise identifier
+    Upload collection instrument
+    Upload a custom spreadsheet and insert into encrypted DB column
+    :param id: Survey identifier
     :type id: str
-    :param count: Number of items in batch
-    :type count: int
+    :param file: File name
+    :type file: str
+    :param files: The file to upload
+    :type files: werkzeug.datastructures.FileStorage
 
     :rtype: None
     """
-    return 'do some magic!'
+    uploaded_files = None
+    for index in ['files', 'files[]', 'upfile']:
```

Comment:
Only one will be supplied, this isn't new code, not sure why it's popping up here.